otherlibs/dune-site/test/dune

```diff
@@ -4,13 +4,10 @@
   (package dune)
   (package dune-site)))
 
-; The test being broken on CI, we deactivate it when
-; we detect that the CI environment variable is not
-; set. Most CI systems set it.
-
 (cram
  (applies_to run_2_9 run)
  (enabled_if
   (and
    (<> %{os_distribution} nixos)
-   (not %{env:CI=false}))))
+   (not %{env:CI=false})
+   (not %{env:NIX_CI=false}))))
```

Comment:
Why do you introduce this new variable? Why don't you just set CI in nix?

Reply:
There are other tests that I am disabling only for nix, but these are still being run elsewhere in the CI.

Reply:
We have four different cases:

1. Nix
2. CI
3. Nix + CI
4. Not nix and not CI

I claim that two variables should be enough to express all of these possibilities. If you disagree, could you give me an example?

Reply:
I see. Still this solution strikes me as incomplete. What about users on ubuntu + nix running the tests locally?

Do we have another variable that could be helpful? `os_family` perhaps? If not, we could consider introducing something ourselves here. In the worst case, we could always use some existing variable that nix sets. There should be plenty to choose from.

Reply:
There isn't a reliable way to check if we are inside a "nix shell". There is "IN_NIX_SHELL" which isn't set reliably between nix commands. Therefore, I've instead introduced a "INSIDE_NIX" variable that is set in each of the dev shell derivations. That way we only need a single variable for checking for nix. We aren't concerned whether we are on nixos, since that isn't the part that affects many of the tests.

Reply:
Okay thanks. This is good enough for now.